build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -520,6 +520,49 @@ message Platform {
   // The server MAY use the `value` of one or more properties to determine how
   // it sets up the execution environment, such as by making specific system
   // files available to the worker.
+  //
+  // The following standard `name`s are defined:
+  //
+  // * `OS`: This describes the operating system. Only a single value is allowed
+  //   and an exact match is required. The following standard values are
+  //   defined:
+  //
+  //     - AIX
+  //     - FreeBSD
+  //     - Linux
+  //     - MacOS
+  //     - SunOS
+  //     - Windows
+  //
+  // * `ISA`: This describes the instruction set architecture including
+  //   instruction set extensions. Multiple values are allowed. The worker is
```

Comment:
I don't see a good option how we could fully address this with the current string-based key-value structure. I'd consider this an edge case we could skip for now.

Either the FreeBSD worker could claim to be a Linux-only system or if running a mix of FreeBSD and Linux binaries is desired, the Linux compatibility layer support could be specified in a separate platform property (just like other kernel/OS config options).

Or do you have another idea how to cover all of this nicely?

Reply:
Are you happy not covering such edge cases with the standard platform lexicon for the time being?

Reply:
Sure, but I think this could be solved relatively easy, though. What if the OS+ISA were announced through a single label (key or value)? "I want the build to run on a system that supports FreeBSD/amd64." This is, in fact, also closer towards the way OS kernels model their ISA support. For example, on FreeBSD, every OS+ISA pair has its own kernelspace `struct sysentvec` that points to callbacks on how to deal with syscalls, faults, signals, etc.

Reply:
This wouldn't allow adding more detailed properties such as `OSKernelVersion` or `OSDistribution` in the future, and already ISA extension handling might require extra effort. I.e., we'd essentially have to ignore the key/value structure for anything OS/ISA-related and define a new text-based format that covers these parts. That's definitely possible but seems like a workaround, not how the spec was designed.

Clients issuing actions that require execution on multiple operating system ABIs seems pretty niche to me. Workers that support multiple operating system ABIs might be interesting in more cases but maybe the better approach for that would be to allow workers to advertise multiple `Platform` instances, each with `OSFamily` and `ISA`.